File: array-3427.cpp
#include <iostream>
#include <vector>
#include <cmath>

int main() {
    std::vector<int> nums = {2,3,1};
    int count = 0;

    for (int i = 0; i < nums.size(); i++) {
        int start = max(0, i - nums[i]);

        for (int j = start; j <= i; j++) {
            count += nums[j];
        }
    }

    std::cout << count;
}

File: array-3427.py
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9]
n = len(nums)
s = 0

for i in range(n):
    start = max(0, i - nums[i])
    end = i+1
    s += sum(nums[start:end])
print(s)